Disable Envoy version check if L7 proxy is disabled

bmcustodio opened the pull request:
If `--enable-l7-proxy` is set to `false`, there is no point in performing the Envoy version check, and so it should be automatically disabled. One of the motivations for this change is that in some scenarios it's necessary to disable the L7 proxy to allow Cilium to start (eg. https://github.com/cilium/cilium/issues/17467), but specifying `--enable-l7-proxy=false` is not enough due to the version check still being performed in this situation. This commit attempts to improve the UX by disabling the version check if Envoy itself is disabled.

```release-note
Envoy version checking is now disabled whenever L7 proxy is disabled too
```

pchaigno commented on pkg/option/config.go:
```diff
@@ -3226,6 +3225,12 @@ func (c *DaemonConfig) Populate() {
+        c.DisableEnvoyVersionCheck = viper.GetBool(DisableEnvoyVersionCheck)
+        if !c.EnableL7Proxy {
+          c.DisableEnvoyVersionCheck = true
+        }
```
Indentation is wrong here: spaces instead of tabs.

bmcustodio committed:
Disable Envoy version check if L7 proxy is disabled

If '--enable-l7-proxy' is set to 'false', there is no point in
performing the Envoy version check, and so it should be automatically
disabled. One of the motivations for this change is that in some
scenarios it's necessary to disable the L7 proxy to allow Cilium to
start (eg. #17467), but
specifying '--enable-l7-proxy=false' is not enough due to the version
check still being performed in this situation. This commit attempts to
improve the UX by disabling the version check if Envoy itself is
disabled.

Signed-off-by: Bruno M. Custódio <brunomcustodio@gmail.com>

pippolo84 reviewed (approved):
LGTM
